src/main/java/uk/gov/hmcts/reform/lrdapi/util/ValidationUtils.java

```diff
@@ -203,4 +211,56 @@ private static void checkIfStringStartsAndEndsWithComma(String csvIds, String ex
         }
     }
 
+    /**
+     * A util method to check additional filter validation.
+     *
+     * @param requestParam Additional parameters to validate.
+     */
+    public static void validateCourtVenueFilters(CourtVenueRequestParam requestParam) {
+        var hearingLocation = requestParam.getIsHearingLocation();
+        if (hearingLocation != null
+            && !StringUtils.equalsIgnoreCase(hearingLocation, IS_Y)
+            && !StringUtils.equalsIgnoreCase(hearingLocation, IS_N)) {
+            throw new InvalidRequestException(String.format(INVALID_ADDITIONAL_FILTER, FILTER_IS_HEARING_LOCATION));
+        }
+
+        var caseManagementLocation = requestParam.getIsCaseManagementLocation();
+        if (caseManagementLocation != null
+            && !StringUtils.equalsIgnoreCase(caseManagementLocation, IS_Y)
+            && !StringUtils.equalsIgnoreCase(caseManagementLocation, IS_N)) {
+            throw new InvalidRequestException(String.format(INVALID_ADDITIONAL_FILTER,
+                                                            FILTER_IS_CASE_MANAGEMENT_LOCATION));
+        }
+
+        if (ObjectUtils.isNotEmpty(requestParam.getLocationType())) {
+            checkSpecialCharacters(requestParam.getLocationType());
+        }
+
+        var temporaryLocation = requestParam.getIsTemporaryLocation();
+        if (temporaryLocation != null
+            && !StringUtils.equalsIgnoreCase(temporaryLocation, IS_Y)
+            && !StringUtils.equalsIgnoreCase(temporaryLocation, IS_N)) {
+            throw new InvalidRequestException(String.format(INVALID_ADDITIONAL_FILTER,
+                                                            FILTER_IS_TEMPORARY_LOCATION));
+        }
+    }
+
+    public static CourtVenueRequestParam trimCourtVenueRequestParam(CourtVenueRequestParam requestParam) {
+        var result = new CourtVenueRequestParam();
+
+        if (ObjectUtils.isNotEmpty(requestParam.getIsHearingLocation())) {
+            result.setIsHearingLocation(StringUtils.strip(requestParam.getIsHearingLocation()));
```

Comment:
1) instead of 4 if statements you can use ternary operator like below to reduce no of linea

result.setIsHearingLocation(ObjectUtils.isNotEmpty(requestParam.getIsHearingLocation())?StringUtils.strip(requestParam.getIsHearingLocation():null)

2) Also try to use static imports for ObjectUtils and StringUtils

Reply:
Done !!